dask/dataframe/io/delta/core.py

```diff
@@ -0,0 +1,289 @@
+                if row["metaData.schemaString"]:
+                    self.schema = schema_from_string(row["metaData.schemaString"])
+                elif row["add.path"]:
+                    self.pq_files.add(f"{self.path}/{row['add.path']}")
+
+    def get_pq_files_from_delta_json_logs(self):
+        """
+        start from checkpoint id, collect logs from every json file until the
+        given version
+        example:
+            checkpoint 10, version 16
+                1. read the logs from 10th checkpoint parquet ( using above func)
+                2. read logs from json files until version 16
+        log Collection:
+            for reading the particular version of delta table, We are concerned
+            about `add` and `remove` Operation (transaction) only.(which involves
+            adding and removing respective parquet file transaction)
+        """
+        log_files = self.fs.glob(
+            f"{self.delta_log_path}/{self.checkpoint // 10:019}*.json"
+        )
+        if len(log_files) == 0:
+            raise RuntimeError(
+                f"No Json files found at _delta_log_path:- {self.delta_log_path}"
+            )
+        log_files = sorted(log_files)
+        log_versions = [
+            int(re.findall(r"(\d{20})", log_file_name)[0])
+            for log_file_name in log_files
+        ]
+        if (self.version is not None) and (self.version not in log_versions):
+            raise ValueError(
+                f"Cannot time travel Delta table to version {self.version}, Available versions for given "
+                f"checkpoint {self.checkpoint} are {log_versions}"
+            )
+        for log_file_name, log_version in zip(log_files, log_versions):
+            with self.fs.open(log_file_name) as log:
+                for line in log:
+                    meta_data = json.loads(line)
+                    if "add" in meta_data.keys():
+                        file = f"{self.path}/{meta_data['add']['path']}"
+                        self.pq_files.add(file)
+
+                    elif "remove" in meta_data.keys():
+                        remove_file = f"{self.path}/{meta_data['remove']['path']}"
+                        if remove_file in self.pq_files:
+                            self.pq_files.remove(remove_file)
+                    elif "metaData" in meta_data.keys():
+                        schema_string = meta_data["metaData"]["schemaString"]
+                        self.schema = schema_from_string(schema_string)
+
+                if self.version == int(log_version):
+                    break
+
+    def read_delta_dataset(self, f, **kwargs):
+        """
+        Function to read single parquet file using pyarrow dataset (to support
+        schema evolution)
+        """
+        schema = kwargs.pop("schema", None) or self.schema
+        filter = kwargs.pop("filter", None)
+        return (
+            pa_ds.dataset(
+                source=f,
+                schema=schema,
```

Comment:
Is providing a schema here the critical thing, to enable evolution?

Reply:
I think yes, different versions of delta files may have different schemas, and using schema parameter in pyarrow dataset we are controlling it, as well as User can explicitly pass the `pyarrow.schema` , Anything needs to be changed here?